service/src/main/java/bio/terra/drshub/services/BardApiFactory.java

```diff
@@ -0,0 +1,94 @@
+package bio.terra.drshub.services;
+
+import bio.terra.bard.api.BardApi;
+import bio.terra.bard.client.ApiClient;
+import bio.terra.bard.model.EventProperties;
+import bio.terra.common.iam.BearerToken;
+import bio.terra.drshub.DrsHubException;
+import bio.terra.drshub.config.DrsHubConfig;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.List;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public record BardApiFactory(DrsHubConfig drsHubConfig) {
```

Comment:
Is this a place where we'd benefit from sharing common HTTP clients?  I wasn't 100% sure if it was relevant here.  We made that switch in TDR and it helped our memory usage quite a bit:

https://github.com/DataBiosphere/jade-data-repo/pull/1462

Reply:
Hmmm maybe.  Let me look

Reply:
It can be treacherous to share api clients: https://github.com/DataBiosphere/terra-drs-hub/pull/75

Reply:
Especially since each API Client has the bearer-token baked into it, and it can't be set per-request.

Reply:
Yeah, I'll leave it separate